modules/core/src/smithy4s/Service.scala

```diff
@@ -34,7 +34,7 @@ import kinds._
   *   around makes it drastically easier to implement logic generically, without involving
   *   metaprogramming.
   */
-trait Service[Alg[_[_, _, _, _, _]]] extends FunctorK5[Alg] with HasId {
+trait Service[Alg[_[_, _, _, _, _]]] extends FunctorK5[Alg] with HasId with Mirror[Alg] {
```

Comment:
I'm not sure this coupling is desirable at this point, so let's not bake it in (the companion object of generated types can however implement both interfaces)

Reply:
hmm, that might solve our next question 😄 , we were not sure about how to implement this for the compliance test service that's generated with a single endpoint. I removed this `with Mirror` at e2d12aa1983e1103b64f8096bc240529998357b7 , but have kept the changes in the dynamic module, let us know if those make sense

Reply:
@Baccata I remember this being useful for being able to find the "static" variant after the user gives you the "normal" one (e.g. `MyInterpreter(MyService)` having easy access to `MyServiceStatic`).

Reply:
like, either of the examples in the original issue?

```scala
endpointBuilder(WeatherService)
  .getWeather { input => IO.println(input) }
  //.postWeather - this method is not implemented, so it won't be handled by the implementation
  .build
```

```scala
gatlingBuilder(WeatherService)
  .withEndpoint(_.getWeather(session => session("city")))
```

Reply:
if it can be done with `extends Mirror` then disregard me :P 

but I think it would be best if users of interpreters don't have to do `endpointBuilder(WeatherServiceProduct)`.

Reply:
I can deal with that.

probably no use for `Mirror` in Dynamic world either, so it makes sense.

Reply:
at [2cba2be](https://github.com/disneystreaming/smithy4s/pull/1011/commits/2cba2bec54ab74b4d7911aa37b4a7e6428111e09) I moved the extends to generated code. I haven't removed the extends from Dynamic , let me know if there's no use case there either and will remove that

Reply:
removed the dynamic changes at [95fc8c3](https://github.com/disneystreaming/smithy4s/pull/1011/commits/95fc8c38a17369f9f979a3849578faf2912df212)